### PR TITLE
Add event generator

### PR DESCRIPTION
**Pull request by xv2**

### CODE REVIEW

File: eventGenerator/generator.js [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+import storageDriver from 'resolve-storage-file';
+
+import eventTypes from '../common/events/index';
+import HNServiceRest from './services/HNServiceRest';
+
+const MAX_STORY_COUNT = 50;
+
+let events = [];
+let users = {};
+let storyIds = [];
```
Comment (timbset) [inline eventGenerator/generator.js]: Use can use `const` for events, users and storyIds

File: eventGenerator/generator.js [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+
+const dbPath = './storage.json';
+let canceled = false;
+
+const isCanceled = () => canceled;
```
Comment (timbset) [inline eventGenerator/generator.js]: Is it really necessary to use function but not a variable?

File: eventGenerator/generator.js [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+  events.push({
+    type,
+    aggregateId,
+    timestamp,
+    payload
+  });
+
+const generateUserEvents = name => {
+  const aggregateId = uuid.v4();
+  addEvent(USER_CREATED, aggregateId, new Date(3600 * 24 * 1000).getTime(), {
```
Comment (timbset) [inline eventGenerator/generator.js]: `new Date(3600 * 24 * 1000).getTime()` === `3600 * 24 * 1000`. P.S. Extract to the const

File: eventGenerator/generator.js [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+      Promise.resolve()
+    )
+  );
+};
+
+const getStories = path =>
+  HNServiceRest.storiesRef(path).then(res => res.json());
+
+export const start = (countCallback, tickCallback) => {
+  if (fs.existsSync(dbPath)) fs.unlinkSync(dbPath);
```
Comment (timbset) [inline eventGenerator/generator.js]: It's better to add braces

File: eventGenerator/generator.js [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+  canceled = false;
+  return Promise.all([
+    getStories('topstories'),
+    getStories('newstories'),
+    getStories('beststories'),
+    getStories('askstories'),
+    getStories('showstories'),
+    getStories('jobstories')
+  ])
+    .then(categories => {
```
Comment (timbset) [inline eventGenerator/generator.js]: Try to use async/await. It's more modern and readable

File: eventGenerator/generator.js [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+      let stories = categories.reduce(
```
Comment (timbset) [inline eventGenerator/generator.js]: Use const

File: eventGenerator/services/HNService.js [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+require('firebase/database');
+
+var config = {
+  databaseURL: 'https://hacker-news.firebaseio.com'
+};
+firebase.initializeApp(config);
+var version = '/v0';
+var api = firebase.database().ref(version);
+
+// https://firebase.google.com/support/guides/firebase-web
```
Comment (timbset) [inline eventGenerator/services/HNService.js]: Is it necessary comment? And why do you use `var`? `const` and `let` are supported by the latest node.js

**Review by timbset — CHANGES_REQUESTED** [review]

File: eventGenerator/generator.js (hunk 1 above)
Comment (xv2) [reply eventGenerator/generator.js]: fixed

File: eventGenerator/generator.js (hunk 2 above)
Comment (xv2) [reply eventGenerator/generator.js]: fixed

File: eventGenerator/generator.js (hunk 3 above)
Comment (xv2) [reply eventGenerator/generator.js]: fixed

File: eventGenerator/generator.js (hunk 4 above)
Comment (xv2) [reply eventGenerator/generator.js]: fixed

File: eventGenerator/generator.js (hunk 6 above)
Comment (xv2) [reply eventGenerator/generator.js]: fixed

**Review by timbset — APPROVED** [review]

**Review by MrCheater — APPROVED** [review]

**Pull request merged**